version: '3.8'

services:
  api:
    image: matheushpr9/ordine-api-app:v1.0.3
    command: go run main.go run
    volumes:
      - .:/app
    depends_on:
      - database
    environment:
      LOG_LEVEL: INFO
      ALLOWED_ORIGINS: localhost,https://test.com
    ports:
      - "8080:8080" # Posicione fora de deploy
    deploy:
      mode: replicated
      replicas: 2

      restart_policy:
        condition: on-failure

  database:
    image: matheushpr9/ordine-api-database:v1.0.1
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: bob_live
      MYSQL_PASSWORD: pwdlocal
      MYSQL_DATABASE: ordine
      TZ: "America/Sao_Paulo"
    volumes:
      - database:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-uroot", "-proot"]
      interval: 10s
      retries: 30
      start_period: 60s
      timeout: 5s
    ports:
      - "3306:3306" # Posicione fora de deploy
    deploy:
      mode: replicated
      replicas: 1
      resources:
        limits:
          cpus: "1.0"
          memory: "500M"
      restart_policy:
        condition: on-failure

  visualizer:
    image: dockersamples/visualizer:latest
    environment:
      - DOCKER_HOST=tcp://docker:2375  # Altere para o endere√ßo correto do Docker
      - HOST=tasks.visualizer
      - PORT=8080
    ports:
      - "8081:8080" # Posicione fora de deploy
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints: [node.role == manager]
      resources:
        limits:
          cpus: "0.1"
          memory: "50M"

volumes:
  database: {}
